File: Encapsulamiento/Java/Persona.java
public class Persona{
    private String curp;
    public String nombre;
    public String apellido;
    public int edad;

    public Persona(String name, String last, int age, String cc){
        this.nombre = name;
        this.apellido = last;
        this.edad = age;
        this.curp = cc;
    }

    public void caminar(){
        System.out.println("Estoy caminando");
    }

    public void dormir(){
        System.out.println("zzz");
    }

    public void hablar(){
        System.out.println("Hola, yo soy "+this.nombre);
    }

    String getCURP(){   // getter
        return curp;
    }
    
    void setCURP(String newCurp){   // setter
        this.curp = newCurp;
    }
}

File: Encapsulamiento/python/Persona.py
class Persona(object):
    def __init__(self,name,last,age,cc):
        self.nombre = name
        self.apellido = last
        self.edad = age
        self._curp = cc

    def hablar(self):
        print("Hola, soy {}".format(self.nombre))

    def caminar(self):
        print("Estoy caminando")

    def dormir(self):
        print("zzz")

    def getCURP(self):
        return self._curp

    def setCURP(self,newCurp):
        self._curp = newCurp

    def __del__(self):
        print("Me mori Dx")